crates/rethnet_evm/src/db/remote.rs

```diff
@@ -0,0 +1,86 @@
+use tokio::runtime::{Builder, Runtime};
+
+use rethnet_eth::remote::{RpcClient, RpcClientError};
+use rethnet_eth::{Address, B256, U256};
+
+use revm::{db::DatabaseRef, AccountInfo, Bytecode};
+
+pub struct RemoteDatabase {
+    client: RpcClient,
+    runtime: Runtime,
+}
+
+#[derive(thiserror::Error, Debug)]
+pub enum RemoteDatabaseError {
+    #[error("The requested method does not have an implementation")]
+    UnimplementedMethod(),
+
+    #[error(transparent)]
+    RpcError(#[from] RpcClientError),
+
+    /// Some other error from an underlying dependency
+    #[error(transparent)]
+    OtherError(#[from] std::io::Error),
+}
+
+impl RemoteDatabase {
+    pub fn _new(url: &str) -> Self {
```

Comment:
```suggestion
    pub fn new(url: &str) -> Self {
```

Reply:
You can avoid the warning by adding it to the public API of the crate

Reply:
Addressed in https://github.com/NomicFoundation/hardhat/pull/3546/commits/a45659fcf4ce23da81ed14fbf3526117bb668b6a